storage/src/cache/dedup/db.rs

```diff
@@ -0,0 +1,329 @@
+// Copyright (C) 2022-2023 Alibaba Cloud. All rights reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(unused)]
+
+use std::path::Path;
+
+use r2d2::{Pool, PooledConnection};
+use r2d2_sqlite::SqliteConnectionManager;
+use rusqlite::{Connection, DropBehavior, OptionalExtension, Transaction};
+
+use super::Result;
+
+pub struct CasDb {
+    pool: Pool<SqliteConnectionManager>,
+}
+
+impl CasDb {
+    pub fn new(path: impl AsRef<Path>) -> Result<CasDb> {
+        let mut db_path = path.as_ref().to_owned();
+        db_path.push("cas.db");
+        Self::from_file(db_path)
+    }
+
+    pub fn from_file(db_path: impl AsRef<Path>) -> Result<CasDb> {
+        let mgr = SqliteConnectionManager::file(db_path);
+        let pool = r2d2::Pool::new(mgr)?;
+        let conn = pool.get()?;
+
+        conn.execute(
+            "CREATE TABLE IF NOT EXISTS Blobs (
+            BlobId  INTEGER PRIMARY KEY,
+            FilePath   TEXT NOT NULL UNIQUE
+        )",
+            (),
+        )?;
+        /*
+        conn.execute(
+            "CREATE INDEX IF NOT EXISTS BlobIndex ON Blobs(FilePath)",
+            (),
+        )?;
+         */
+
+        conn.execute(
+            "CREATE TABLE IF NOT EXISTS Chunks (
+            ChunkId           TEXT NOT NULL,
+            ChunkOffset       INTEGER,
+            BlobId            INTEGER,
```

Comment:
Take `4685855ca4467b60347b04bacc810ce80f96e164c9e131715e585f0be2ab01f7` as example. Not sure whether large enough. Maybe text is better.